src/passkey_child/passkey_child.h

```diff
@@ -170,6 +173,45 @@ generate_credentials(struct passkey_data *data, fido_dev_t *dev,
 errno_t
 verify_credentials(const fido_cred_t *const cred);
 
+/**
+ * @brief Format and write public key in PEM format
+ *
+ * @param[in] file File stream in which to write
+ * @param[in] ptr Public key
+ * @param[in] len Public key length
+ *
+ * @return 0 if the key was formatted and written properly,
+ *         another value on error.
+ */
+errno_t
+write_es256_pubkey(FILE *file, const void *ptr, size_t len);
```

Comment:
Hi,

I think it would be more flexible to separate the conversion and the writing of the key. So instead of e.g. `write_es256_pubkey()` have something like `es256_pubkey_to_pem()` which just returns a string with the PEM data. For this you have to use `PEM_write_bio_PUBKEY()` on the OpenSSL level instead of `PEM_write_PUBKEY()` and create a `BIO_s_mem` type BIO to write into memory and then `BIO_get_mem_data()` to get the written data, see e.g. [sss_cert_der_to_pem()](https://github.com/SSSD/sssd/blob/master/src/util/cert/libcrypto/cert.c#L27) for details.

With this those functions can be reused by other callers if needed and the actual output can be done in `passkey_child.c` where we know that it is part of a program called by a user.

bye,
Sumit

Reply:
I think I got you, but please check it to make sure

Reply:
Hi,

yes, that's what I meant. Just two comments. First I would suggest that `format_pem_public_key()` returns the string as well so that it is actually printed in `main()`. Second, by moving the `sss_*_pubkey_to_pem()` functions into `SSS_CRYPT` you add a general dependency to `libfido2` to more or less all SSSD components. With "reused by other callers" I just meant that the source file can be included by other components if needed, not that it should be generally available. So I would prefer to keep them in `passkey_child` so that `libfido2` and its dependencies are only pulled in if the package containing `passkey_child` is installed.

bye,
Sumit

Reply:
Done. Thanks for the review.